src/NHibernate.Test/NHSpecificTest/GH0831/FixtureByCode.cs

```diff
@@ -0,0 +1,166 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+
+using NHibernate.Cfg.MappingSchema;
+using NHibernate.Mapping.ByCode;
+
+using NUnit.Framework;
+
+namespace NHibernate.Test.NHSpecificTest.GH0831
+{
+	public class ByCodeFixture : TestCaseMappingByCode
+	{
+		private readonly IList<Entity> entities = new List<Entity>
+		{
+			new Entity { EntityValue = 0.5m },
+			new Entity { EntityValue = 1.0m },
+			new Entity { EntityValue = 1.5m },
+			new Entity { EntityValue = 2.0m },
+			new Entity { EntityValue = 2.5m },
+			new Entity { EntityValue = 3.0m }
+		};
+
+		protected override HbmMapping GetMappings()
+		{
+			var mapper = new ModelMapper();
+			mapper.Class<Entity>(rc =>
+			{
+				rc.Id(x => x.Id, m => m.Generator(Generators.GuidComb));
+				rc.Property(x => x.EntityValue);
+			});
+
+			return mapper.CompileMappingForAllExplicitlyAddedEntities();
+		}
+
+		protected override void OnSetUp()
+		{
+			using (ISession session = OpenSession())
+			using (ITransaction transaction = session.BeginTransaction())
+			{
+				foreach (Entity entity in entities)
+				{
+					session.Save(entity);
+				}
+
+				session.Flush();
+				transaction.Commit();
+			}
+		}
+
+		protected override void OnTearDown()
+		{
+			using (ISession session = OpenSession())
+			using (ITransaction transaction = session.BeginTransaction())
+			{
+				session.Delete("from System.Object");
+
+				session.Flush();
+				transaction.Commit();
+			}
+		}
+
+		private void IgnoreIfNotSupported(string function)
```

Comment:
With the merge of #1535 in master, this helper should be removed for using `AssumeFunctionSupported` (inherited from `TestCase`) instead.